### ISSUE TITLE
ObservableTimeTest.Delay_DateTimeOffset_DefaultScheduler stuck in CI

### ISSUE THREAD

**Issue by bartdesmet** [dotnet/reactive#382]
See https://ci.appveyor.com/project/dotnetfoundation/rx-net-la3iv/build/4.0.0-MiscSubjectImprovements.180.build.281

```
[00:59:28] [xUnit.net 00:54:11.7505664] Tests.System.Reactive: [Long Running Test] 'ReactiveTests.Tests.ObservableTimeTest.Delay_DateTimeOffset_DefaultScheduler', Elapsed: 00:53:51
```

**Comment by akarnokd** [comment]
There could be a bug in the operator that under some circumstances stops handling the delayed signals and the test just hangs waiting for the operator to complete. When the test is looped with practically a zero shift, it tends to hang after several hundred rounds on my machine:

```cs
for (int i = 0; i < 10000; i++)
{
    Assert.True(Observable.Return(1)
        .Delay(DateTimeOffset.UtcNow/* + TimeSpan.FromMilliseconds(1)*/)
        .ToEnumerable()
        .SequenceEqual(new[] { 1 }));
}
```

**Comment by akarnokd** [comment]
Fix posted in #560.
